support/cas-server-support-cassandra-ticket-registry/src/main/java/org/apereo/cas/ticket/registry/CassandraTicketRegistry.java

```diff
@@ -0,0 +1,254 @@
+package org.apereo.cas.ticket.registry;
+
+import com.datastax.driver.core.Cluster;
+import com.datastax.driver.core.ConsistencyLevel;
+import com.datastax.driver.core.PreparedStatement;
+import com.datastax.driver.core.ProtocolVersion;
+import com.datastax.driver.core.ResultSet;
+import com.datastax.driver.core.Row;
+import com.datastax.driver.core.Session;
+import org.apereo.cas.TicketSerializer;
+import org.apereo.cas.ticket.Ticket;
+import org.apereo.cas.ticket.TicketCatalog;
+import org.apereo.cas.ticket.TicketDefinition;
+import org.apereo.cas.ticket.TicketGrantingTicketImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.ZonedDateTime;
+import java.util.Collection;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+import static org.apereo.cas.config.CassandraTicketRegistryTicketCatalogConfiguration.ST_TABLE;
+import static org.apereo.cas.config.CassandraTicketRegistryTicketCatalogConfiguration.TGT_TABLE;
+
+/**
+ * @author David Rodriguez
+ *
+ * @since 5.2.0
+ */
+public class CassandraTicketRegistry<T> extends AbstractTicketRegistry implements CassandraTicketRegistryDao {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CassandraTicketRegistry.class);
+
+    private static final int FIRST_COLUMN_INDEX = 0;
+    private static final long TEN_SECONDS = 10000L;
+    private static final int TEN = 10;
+
+    private final TicketCatalog ticketCatalog;
+    private final TicketSerializer<T> serializer;
+    private final Class<T> typeToWriteToCassandra;
+
+    private final PreparedStatement insertTgtStmt;
```

Comment:
Prepared statements per ticket type need to be dynamically constructed and cached if need be.